pkg/da/data_availability_header.go

```diff
@@ -167,6 +167,14 @@ func (dah *DataAvailabilityHeader) IsZero() bool {
 	return len(dah.ColumnRoots) == 0 || len(dah.RowRoots) == 0
 }
 
+// SquareSize returns the number of rows in the original data square.
+func (dah *DataAvailabilityHeader) SquareSize() (int, error) {
+	if len(dah.RowRoots)%2 != 0 {
+		return 0, fmt.Errorf("len row roots is not divisible by 2: %d", len(dah.RowRoots))
+	}
+	return len(dah.RowRoots) / 2, nil
```

Comment:
Are we sure it's worth checking %2 and doing err != nil everywhere?

Reply:
Defer to you b/c AFAIK there are no other usecases for this helper outside celestia-node.

Removed return parameter in https://github.com/celestiaorg/celestia-app/pull/2095/commits/8ef4bd6158cf8338f92a44f87e57a2cfcd80254b